src/main/java/com/iexec/sms/secret/compute/TeeTaskComputeSecret.java

```diff
@@ -21,14 +21,27 @@
 import lombok.NoArgsConstructor;
 import org.hibernate.annotations.GenericGenerator;
 
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.Id;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
 
+/**
+ * Define a secret that can be used during the execution of a TEE task.
+ * Currently, only secrets for application developers and requesters are supported.
+ * <p>
+ * In this implementation, a unique constraint has been added on <b>onChainObjectAddress</b>,
+ * <b>fixedSecretOwner</b> and <b>key</b> columns.
+ * This constraint has been defined in such a way because:
+ * <ul>
+ * <li>For application developers, fixedSecretOwner will always be "". Each application developer
+ *     secret is uniquely identified with onChainObjectAddress and key (long parsed as String) values.
+ * <li>For requesters, onChainObjectAddress will always be "". Each requester secret is uniquely
+ *     identified with fixedSecretOwner and key values.
+ * </ul>
+ */
 @Data
 @NoArgsConstructor
 @Entity
+@Table(uniqueConstraints = { @UniqueConstraint(columnNames = {"onChainObjectAddress", "fixedSecretOwner", "key"}) })
```

Comment:
I'm wondering whether this should include `secretOwnerRole` as well. If we later add beneficiaries, that could break current constraint.

Reply:
You have a point, I think we should wait for a clear feature request on this point before changing this.